src/utils/Auth.js

```diff
@@ -0,0 +1,22 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { getInstance } from 'd2/lib/d2';
+
+
+export const withAuth = (Component) => {
+
+    const Auth = (props, { d2 }) => {
+        const extraProps = {
+            getCurrentUser: () => d2.currentUser,
```

Comment:
I think there wasn't really a reason why this was a function.  It might as well just be `currentUser`.

Reply:
Only reason I kept it like this was to keep the "API" the same. As the old mixin used a function, but I agree.

Reply:
Yeah i thought as much, totally makes sense. My comment was more a general observation :P